lib/runner/extract-runnable-items.js

```diff
@@ -0,0 +1,141 @@
+var sdk = require('postman-collection'),
+    ItemGroup = sdk.ItemGroup,
+    Item = sdk.Item,
+
+    DEFAULT_LOOKUP_STRATEGY = 'idOrName',
+    INVALID_LOOKUP_STRATEGY_ERROR = 'runtime~extractRunnableItems: Invalid entrypoint lookupStrategy',
+
+    /**
+     * Accumulate all items in order if entry point is a collection/folder.
+     * If an item is passed returns an array with that item.
+     *
+     * @param {ItemGroup|Item} node
+     *
+     * @returns {Array<Item>}
+     *
+     * @todo: Possibly add mapItem to sdk.ItemGroup?
+     */
+    flattenNode = function (node) {
+        var items = [];
+
+        // bail out
+        if (!node) { return items; }
+
+        if (ItemGroup.isItemGroup(node)) {
+            node.forEachItem(function (item) { items.push(item); });
+        }
+        else if (Item.isItem(node)) {
+            items.push(node);
+        }
+
+        return items;
+    },
+
+    /**
+     * Finds an item or item group based on id or name.
+     *
+     * @param {ItemGroup} itemGroup
+     * @param {?String} match
+     *
+     * @returns {Item|ItemGroup|undefined}
+     */
+    findItemOrGroup = function (itemGroup, match) {
+        if (!itemGroup || !itemGroup.items) { return; }
+
+        var matchNode;
+
+        // lookup match on own children
+        itemGroup.items.each(function (itemOrGroup) {
+            if (itemOrGroup.id === match || itemOrGroup.name === match) {
+                matchNode = itemOrGroup;
+                return false; // exit the loop
+            }
+        });
+
+        // if there is no match on own children, start lookup on grand children
+        !matchNode && itemGroup.items.each(function (itemOrGroup) {
+            matchNode = findItemOrGroup(itemOrGroup, match);
+            if (matchNode) { return false; } // exit the loop
+        });
+
+        return matchNode;
+    },
+
+
+    /**
+     * Finds an item or group from a path. The path should be an array of ids from the parent chain.
+     *
+     * @param {Collection} collection
+     * @param {Object} options
+     * @param {Function} callback
+     */
+    lookupByPath = function (collection, options, callback) {
```

Comment:
we could add docs for possible attributes for `options`
 - execute
 - path

Reply:
Yeah, good point